README.md

```diff
@@ -188,10 +188,15 @@ third argument to `Validator::validate()`, or can be provided as the third argum
 | `Constraint::CHECK_MODE_COERCE_TYPES` | Convert data types to match the schema where possible |
 | `Constraint::CHECK_MODE_APPLY_DEFAULTS` | Apply default values from the schema if not set |
 | `Constraint::CHECK_MODE_EXCEPTIONS` | Throw an exception immediately if validation fails |
+| `CHECK_MODE_EARLY_COERCE` | Apply type coercion as soon as possible |
 
 Please note that using `Constraint::CHECK_MODE_COERCE_TYPES` or `Constraint::CHECK_MODE_APPLY_DEFAULTS`
 will modify your original data.
 
+`CHECK_MODE_EARLY_COERCE` has no effect unless used in combination with `CHECK_MODE_COERCE_TYPES`. If
+enabled, type coercion will occur as soon as possible, even if the value may already be of another valid
+type.
+
```

Comment:
Could you maybe expand on this a bit to make it more clear? I know what you're talking about, but I'm not sure that someone else coming to the project for the first time would.